### PR TITLE
Eugene/tokio

### PR DESCRIPTION
**Pull request by eugene-chen**
Instead of manually spinning up threads for each task, delegate to tokio.

Replace `std::sync::mpsc`` channels with `tokio::sync::mpsc` channels.

Removes anti-pattern of passing around a `tokio::runtime::Runtime` and calling `runtime.block_on`.

Remove Binance listener which doesn't play nicely with this paradigm (bc of the Binance crate we're using).

### CODE REVIEW

File: rust/Cargo.toml [hunk 1]
```diff
@@ -21,8 +21,9 @@ solana-sdk = "1.10.32"
 borsh = "0.9.3"
 rand = "0.7.3"
 phoenix-types = { git = "https://github.com/Ellipsis-Labs/phoenix-types", branch = "master" }
-ellipsis-client = "0.1.10"
-tokio = { version = "1.8.4", features = ["full"] }
+# ellipsis-client = "0.1.10"
+ellipsis-client = { path = "../../ellipsis-client" }
```
Comment (eugene-chen) [inline rust/Cargo.toml:25]: remove before merge

File: rust/phoenix-sdk/src/event_poller.rs [hunk 2]
```diff
@@ -74,23 +67,27 @@ impl EventPoller {
...
                 //       We should be able to spin up chunks of requests and join.
-                let events = rt
-                    .block_on(sdk.parse_events_from_transaction(&signature))
-                    .unwrap_or_default();
-                if event_sender
+                let events = self
+                    .sdk
+                    .parse_events_from_transaction(&signature)
+                    .await
+                    .unwrap();
```
Comment (jarry-xiao) [inline rust/phoenix-sdk/src/event_poller.rs:74]: `unwrap_or_default()`?

File: rust/phoenix-sdk/src/sdk_client.rs [hunk 3]
```diff
@@ -305,26 +305,7 @@ impl SDKClient {
...
-                .ok()?;
-            if raw_tx.transaction.meta.as_ref()?.err.is_some() {
-                return None;
-            }
-
-            parse_transaction(raw_tx)
-        } else {
-            self.client.get_transaction(&sig).await.ok()?
-        };
+        let tx = self.client.get_transaction(sig).await.ok()?;
```
Comment (jarry-xiao) [inline rust/phoenix-sdk/src/sdk_client.rs:308]: So much cleaner :)

**Review by jarry-xiao — APPROVED** [review]
Looks good to me, but please fix the unwrap in EventPoller as we really do not want it dying under any circumstance

**Pull request merged**